### DISCUSSION TITLE
关于HDR，缩放，补帧

### DISCUSSION THREAD

**Discussion by ROOOCCCK** [hooke007/mpv_PlayKit#63] [Q&A 问答] [answered]
关于HDR有什么可以调试的吗

**Comment by hooke007** [comment, marked as answer]
profiles.conf里不都写了吗

**Reply by ROOOCCCK** [reply]
只看到一条T-T

**Reply by hooke007** [reply]
什么一条。。

**Reply by hooke007** [reply]
https://github.com/hooke007/MPV_lazy/blob/main/portable_config/profiles.conf

**Reply by ROOOCCCK** [reply]
我以为都在mpv.conf,抱歉

**Reply by hooke007** [reply]
😂😂

https://github.com/hooke007/MPV_lazy/blob/49589654562e51652ace2ffc268b8c986f9139db/portable_config/mpv.conf#L7

**Reply by ROOOCCCK** [reply]
> 😂😂
> 
> https://github.com/hooke007/MPV_lazy/blob/main/portable_config/mpv.conf#L7-L7

刚刚看了下，只有最后一项跟HDR有关的

**Reply by hooke007** [reply]
直出是个特殊项目，我没想好它的归类所以丢在了mpvconf最后

**Reply by ROOOCCCK** [reply]
我想请教下，我想设置启动播放时自动插帧，不需要播放之后再用快捷键去启用它，该如何修改配置文件呢

**Reply by hooke007** [reply]
https://github.com/hooke007/MPV_lazy/blob/7d934694ba2de3880595c40009d5adf2246a3d0c/portable_config/mpv.conf#L180

**Reply by ROOOCCCK** [reply]
![image](https://user-images.githubusercontent.com/79370772/145830869-6000ad40-bfb8-45cb-bf47-6a7551994a82.png)
我也尝试自己操作一遍（抄设置😂😂），目前有个问题关于统计信息这部分，请问显示缩放算法需要如何设置

**Reply by hooke007** [reply]
你当前没有缩放步骤，不会显示不存在的东西

**Reply by ROOOCCCK** [reply]
我跟平常操作是一样的，缩小画面放大画面都尝试过，我所配置的MPV没有显示出当前使用的是什么算法，研究过你的配置，没有找到相关部分

**Reply by hooke007** [reply]
你在用bilinear？

**Reply by ROOOCCCK** [reply]
都是选用ewa lanczossharp的，在你制作的包可以正常查看用的算法，我自己另外弄的就不行了

**Reply by hooke007** [reply]
这图的数据不会错的，显示的就是当前实际使用的算法。
在没有其它信息的前提下我只能判定为你的操作存在问题

**Reply by ROOOCCCK** [reply]
![image](https://user-images.githubusercontent.com/79370772/145836187-750fc891-7fa6-4001-8963-0deb839583f8.png)
这是我配置的包，算法保持跟之前使用的一致，统计数据没显示当前算法，不知道在哪里修改了

**Reply by hooke007** [reply]
这个数据说明你当前连cscale都是bilinear。。。你可以把直接mpvconf发上来，发图没啥用。。

**Reply by ROOOCCCK** [reply]
用文件传输功能，不要直接贴大量文本占楼（你输入文本的下方就是文件传输）

**Reply by hooke007** [reply]
。。。不要自己创建语法
`[xxx]` 这一行以后的内容全都无效化了，因为被你改成了profile（看手册02）

**Reply by ROOOCCCK** [reply]
😂😂好吧，原来问题出在hdr那里，没用#号，才导致后面的都无效了

**Reply by hooke007** [reply]
我找到你抄到参数出处了。。。你要是也跟他一样用linux（wayland）可能还有点参考意义
用在windows上怪怪的，而且本身就很怪

**Reply by ROOOCCCK** [reply]
我想找像madvr上面的噪点伪影振铃的相关设置，所以看到什么就先抄下来，有空就试🤣

**Reply by hooke007** [reply]
https://github.com/hooke007/MPV_lazy/blob/7d934694ba2de3880595c40009d5adf2246a3d0c/portable_config/mpv.conf#L150

画面的改进基本就这几个，此外就是滤镜和着色器

https://mpv.io/manual/master/#gpu-renderer-options

**Reply by ROOOCCCK** [reply]
![image](https://user-images.githubusercontent.com/79370772/146157496-03f53aa4-ebad-4094-8cc0-78467e53c8d6.png)
丢帧过高，建议如何处理呢，CPU GPU占用均不超过75%。我是使用ewa_lanczossharp，补帧165。有一点很奇怪，我屏幕分辨率是2.5K的，播放4K电影，调用的却是cscale算法
![屏幕截图 2021-12-15 171536](https://user-images.githubusercontent.com/79370772/146157953-6b7684f2-4d08-45b7-9882-a63f969b689c.png)
很奇怪的问题

**Reply by hooke007** [reply]
你这个例子整体需要整机性能很强劲，无法在帧间隔内完成一帧的渲染必然掉帧，何况还是补到100+以上的离谱情况
4k降2k输出不需要upscale；
cscale用途是色度平面缩放，与你的输出分辨率无关；
hdr需要tonemapping的性能开销不低；
当前补帧的计算效率不高，目标帧率降低到刷新率的半值看情况

**Reply by ROOOCCCK** [reply]
说错了，是调用的scale放大算法，这不对呀

**Reply by ROOOCCCK** [reply]
关了svp放4k，算法就对了，用的dcale。是不是svp还需要改动设置

**Reply by hooke007** [reply]
哪个svp？manager还是flow
而且你图里并没有调用放大算法

**Reply by ROOOCCCK** [reply]
flow，懒人包内的那个。scale不是放大用吗

**Reply by hooke007** [reply]
如果你指的是和detect hdr peak 连在一起的那个bilinear，那是保留项目。无视他即可
是上游不想修的也没啥影响的bug
https://github.com/mpv-player/mpv/issues/5727

**Reply by ROOOCCCK** [reply]
应该不是bug，跟svp的补帧有关系

**Reply by hooke007** [reply]
。。。。都说了是bug
触发条件很多，即使目标=原生分辨率下依然会存在
我没研究它触发的所有机制，vs的插入更改了图像格式，这可能就是触发条件之一，因为 **没有任何画面影响** ，所以也懒得研究它的具体条件

**Reply by natural-harmonia-gropius** [reply]
svp那个脚本会缩到1080p，不想要的话改大vw和vh

**Reply by hooke007** [reply]
> svp那个脚本会缩到1080p

我这里测了no-config下，即便是你那个vs脚本单插也不会触发二次bilinear的bug。如果他的描述和操作没有问题，那可能是某个参数和vs一起工作引起的（瞎猜的信息不足无法判定）
![Snipaste_2021-12-15_23-41-32](https://user-images.githubusercontent.com/41094733/146218009-c6b64158-08a1-451d-a95d-7f00e022515e.jpg)

**Reply by ROOOCCCK** [reply]
> svp那个脚本会缩到1080p，不想要的话改大vw和vh

![image](https://user-images.githubusercontent.com/79370772/146318396-56140c4a-3315-4099-8931-84033c50e142.png)
你是z指这个参数吗，这个VW   VH是什么意思，会影响什么呢

**Reply by natural-harmonia-gropius** [reply]
对
viewport width/height
如果视频尺寸比vw vh大会被等比缩放到这个大小

**Reply by ROOOCCCK** [reply]
> 

这个参数有意义吗🤣，mpv本身就有画面缩放。还是说画面要先经过svp缩放再经过mpv？

**Reply by hooke007** [reply]
vs的流程在scale之前，减少分辨率可减少补帧所需的计算量

**Reply by natural-harmonia-gropius** [reply]
掉帧可以试试软解不用copyback

**Reply by ROOOCCCK** [reply]
![image](https://user-images.githubusercontent.com/79370772/146478875-b80d82ea-f79e-43df-b668-4cd5b85da713.png)
今天想折腾下SVPFLOW的脚本，结果播放十几秒之后就会自动出现这个红色的框，怎么消除

**Reply by hooke007** [reply]
。。。你问问题不要只发半截，具体步骤？
我只能根据假设和经验判断 —— svp的无授权警告
